internal/querynodev2/handlers.go

```diff
@@ -313,6 +376,11 @@ func (node *QueryNode) searchChannel(ctx context.Context, req *querypb.SearchReq
 		log.Warn("Query failed, failed to get query shard delegator", zap.Error(ErrGetDelegatorFailed))
 		return nil, ErrGetDelegatorFailed
 	}
+	req, err := node.optimizeSearchParams(ctx, req, sd)
```

Comment:
Do we need to clone a req to make it safe, since maybe multi shardleaders want to modify it

Reply:
we removed search request merging so we could use the request itself and save the cost of clone a request